spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/dsl/MongoDbOutboundGatewaySpec.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mongodb.dsl;
+
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.convert.MongoConverter;
+import org.springframework.expression.Expression;
+import org.springframework.integration.dsl.MessageHandlerSpec;
+import org.springframework.integration.mongodb.outbound.MongoDbOutboundGateway;
+
+/**
+ * A {@link MessageHandlerSpec} extension for the MongoDb Outbound endpoint {@link MongoDbOutboundGateway}
+ *
+ * @author Xavier Padró
+ * @since 5.0
+ */
+public class MongoDbOutboundGatewaySpec
+		extends MessageHandlerSpec<MongoDbOutboundGatewaySpec, MongoDbOutboundGateway> {
+
+	private final MongoDbOutboundGateway mongoDbOutboundGateway;
+
+	MongoDbOutboundGatewaySpec(MongoDbFactory mongoDbFactory, MongoConverter mongoConverter) {
+		this.mongoDbOutboundGateway = new MongoDbOutboundGateway(mongoDbFactory);
+		this.mongoDbOutboundGateway.setMongoConverter(mongoConverter);
+		this.mongoDbOutboundGateway.setRequiresReply(true);
+	}
+
+	MongoDbOutboundGatewaySpec(MongoOperations mongoTemplate) {
+		this.mongoDbOutboundGateway = new MongoDbOutboundGateway(mongoTemplate);
+		this.mongoDbOutboundGateway.setRequiresReply(true);
+	}
+
+	public MongoDbOutboundGatewaySpec expectSingleResult(boolean expectSingleResult) {
+		this.mongoDbOutboundGateway.setExpectSingleResult(expectSingleResult);
+		return this;
+	}
+
+	public MongoDbOutboundGatewaySpec query(String query) {
+		this.mongoDbOutboundGateway.setMongoDbQuery(query);
+		return this;
+	}
+
+	public MongoDbOutboundGatewaySpec queryExpression(String queryExpression) {
```

Comment:
I implemented this for DSL configuration.

However, should we also provide XML support for query function? I made some tests and I don't seem to figure out how the `MongoDbOutboundGatewayParser` should parse the XML property. 

Something like this is not supported by the SpEL parser:

`query-function="msg -> new BasicQuery('{''name'' : msg.payload}')"`

This should be mapped to the `Function<Message<?>, Query> queryFunction` property in the outbound gateway.